public/app/core/components/Login/LoginPage.tsx

```diff
@@ -41,7 +42,7 @@ export const LoginPage: FC = () => {
                   />
                 ) : null}
 
-                {isOauthEnabled ? (
+                {isOauthEnabled || isSamlEnabled ? (
```

Comment:
This logic seems unnecessary. if we moved these wrapping elements inside the LoginServiceButtons component then it can just check if any login service are enabled. It is already dealing with this logic and would clean up and simplify the code on this level 

https://github.com/grafana/grafana/blob/78e823a830369fc9d553718dbb92c446638820f4/public/app/core/components/Login/LoginServiceButtons.tsx#L66

Reply:
Originally, I thought that was the way to go but I kind of like the proposed approach (seems simpler?) in the PR - Moving the wrapping elements inside `LoginServiceButtons` would imply that we now have logic regarding login form itself (via `disableLoginForm`) and overall feels like we have lost the single responsibility approach of `LoginServiceButtons` where we're no longer _only_ rendering buttons.

Whenever we should render the buttons or not feels like the responsibility of this form. Maybe an alternative is giving a name to this logic (`isOauthEnabled || isSamlEnabled`) like `CanShowLoginServiceButtons` or something similar)

I'm not one to argue as your experience in this area overweights mine but thought it was worth mentioning.

Reply:
Feel free to disregard my opinion if you feel strongly about this case, though!

Reply:
yes, I feel like this is more complicated, checking the same thing in multiple ways. If we add another auth provider that is not oauth or saml we would have to add another check here, while if we just checked if we had any available login providers in side the LoginServiceButtons we would not. 

It also makes the LoginServiceButtons a more complete component, it will need a property that said "isLoginFormEnabled" to know if it should add the "or" text at the top but this seems fine to me.

Reply:
I did the exercise to move the wrapping elements inside `LoginServiceButtons ` and I have opened a new PR #19591.

Reply:
So should this PR be closed then or what is the status?

Reply:
@papagian Love your solution on other PR! If @torkelo agrees, I'm happy for us to close this one and merge the other one.

I tested both this PR and #19591 and they both work as expected.